### ISSUE TITLE
crashes with pretty much any of the mods

### ISSUE THREAD

**Issue by youmukonpaku1337** [KenRouKoro/MCMTFabricCE#4]
i removed the threading mods from the list, but it kept crashing until i removed every mod except fabric api and cloth config 
![Screenshot_20220606-205407_JuiceSSH](https://user-images.githubusercontent.com/50289495/172365906-872cd861-1e99-4592-989c-79e1c8577d53.png)

**Comment by youmukonpaku1337** [comment]
no logs because they are pretty much just it crashing with each mod

**Comment by KenRouKoro** [comment]
If you want to use MCMTCE with Lithium, you need to disable the following in the Lithium configuration file.
``` conf
mixin.alloc=false
mixin.entity=false
mixin.collections=false
mixin.chunk=false
mixin.ai.pathing=false
mixin.ai.poi.fast_init=false
```

**Comment by KenRouKoro** [comment]
For MCMTCE, any logs are helpful, including JVM crash reports.

**Comment by KenRouKoro** [comment]
Of course some error reports require you to have a basic knowledge of Chinese (

**Comment by KenRouKoro** [comment]
and does not work with C2ME, please read the README compatibility table carefully.

**Comment by youmukonpaku1337** [comment]
yes, i have removed lithium, dimthread, c2me, and servercore, still crashes

**Comment by youmukonpaku1337** [comment]
it says its incompatible with chunky, paradise lost, collective, and cammie's combat tweaks

**Comment by KenRouKoro** [comment]
Please provide an error log.

**Comment by youmukonpaku1337** [comment]
okay

**Comment by youmukonpaku1337** [comment]
give me a second

**Comment by youmukonpaku1337** [comment]
https://www.toptal.com/developers/hastebin/xahemaziti.sql here is the log

**Comment by KenRouKoro** [comment]
The logs show that Servercore has a Mixin conflict with MCMTCE in TickEntity. I checked the curseforge page for Servercore, which is an optimization module, and like Lithium, its Mixin is fatal for MCMTCE systems. (Servercore also intercepts Entity tick processing)
Please try removing Servercore and retrying.

**Comment by youmukonpaku1337** [comment]
tried that

**Comment by youmukonpaku1337** [comment]
it just errors out about another mod

**Comment by youmukonpaku1337** [comment]
like collective-fabric

**Comment by youmukonpaku1337** [comment]
then cammie's combat tweaks

**Comment by youmukonpaku1337** [comment]
then paradise lost (which i really dont want to remove)

**Comment by KenRouKoro** [comment]
This is why MCMTCE exists. Most modules are not designed to operate in a concurrent environment, and it is normal for errors to occur. To give two examples, AE2 and TECH. the code in AE2 itself is well suited to running concurrently, but the code it uses to determine whether it is in a server environment is implemented by determining whether the current thread is a server thread, but it will only recognise one thread rather than many. MCMTCE determines the thread pool size (i.e. the number of threads) based on the number of CPU cores, and AE2 itself does not recognise this. What I did to fix AE2's compatibility with MCMTCE was to rewrite AE2's server judgement code using Mixin (yes, Mixin can be used for modules and not just Minecraft). Unlike TECH, TECH's cable system does not work in a concurrent environment at all. For its fix, I used Mixin to completely rewrite TECH's cable runtime logic so that its cable system runs in an additional thread, which I call the "Tech cable system daemon thread", using a blocking queue to implement single-threaded logic emulation. In theory, this new system is much better than the original Tech cable system and I am considering submitting it to the Tech project team.
The modules that have been fixed are currently on the compatibility list, I can't really guarantee any compatibility with the modules outside the list. Fixing compatibility is actually complicated, as I need to read the source code of the module and determine the fix, while for closed source modules (such as IC2) it is difficult for me to fix compatibility. This would be a very time consuming task.
The modules you have mentioned have been documented and will be fixed in the future.

**Comment by youmukonpaku1337** [comment]
thats great

**Comment by StockiesLad** [comment]
> This is why MCMTCE exists. Most modules are not designed to operate in a concurrent environment, and it is normal for errors to occur. To give two examples, AE2 and TECH. the code in AE2 itself is well suited to running concurrently, but the code it uses to determine whether it is in a server environment is implemented by determining whether the current thread is a server thread, but it will only recognise one thread rather than many. MCMTCE determines the thread pool size (i.e. the number of threads) based on the number of CPU cores, and AE2 itself does not recognise this. What I did to fix AE2's compatibility with MCMTCE was to rewrite AE2's server judgement code using Mixin (yes, Mixin can be used for modules and not just Minecraft). Unlike TECH, TECH's cable system does not work in a concurrent environment at all. For its fix, I used Mixin to completely rewrite TECH's cable runtime logic so that its cable system runs in an additional thread, which I call the "Tech cable system daemon thread", using a blocking queue to implement single-threaded logic emulation. In theory, this new system is much better than the original Tech cable system and I am considering submitting it to the Tech project team. The modules that have been fixed are currently on the compatibility list, I can't really guarantee any compatibility with the modules outside the list. Fixing compatibility is actually complicated, as I need to read the source code of the module and determine the fix, while for closed source modules (such as IC2) it is difficult for me to fix compatibility. This would be a very time consuming task. The modules you have mentioned have been documented and will be fixed in the future.

You could use fernflower/quiltflower decompiler to get access to its source code. Also, brilliant work with the recode of certain modules from other mods.
